scripts/romanisim-make-stack

```diff
@@ -0,0 +1,267 @@
+                        action='store_true',
+                        help='Use webbpsf for PSFs.')
+
+
+
+
+    args = parser.parse_args()
+
+    log.info('Starting simulation...')
+    log.warning("romanisim is under active development.  Its output has "
+                "not been formally validated; only limited testing has been "
+                "performed.  For this reason, use of romanisim for "
+                "preparation of ROSES proposals is not advised.  Other "
+                "packages like galsim's roman package or STIPS may better "
+                "serve such purposes.")
+
+    # Set up metadata
+    metadata = deepcopy(default_parameters_dictionary)
+
+    # If creating a script, create the script file
+    if args.make_script:
+        spcript_file = open("romansim_stack.script", "w")
+
+        # Acquire cleaned options to pass to script call
+        options_dct = {}
+        for option, value in vars(args).items():
+            if args.debug:
+                print(f'option, value = {option, value}')
+            if option not in ('input_file_name','debug','make_script') and value is not None:
+                options_dct[option] = value
+
+
+
+    with open(args.input_file_name, newline='') as infile:
+        reader = csv.DictReader(infile, delimiter=',', skipinitialspace=True)
+        for entry in reader:
+            # Debug: print keys
+            if args.debug:
+                print(f'entry.keys = {entry.keys()}')
+
+            # Reset time offset
+            time_offset = 0 * u.s
+
+            # RA & Dec
+            coord = SkyCoord(ra=float(entry['ra']) * u.deg, dec=float(entry['dec']) * u.deg,
+                             frame='icrs')
+            wcs.fill_in_parameters(metadata, coord, boresight=args.boresight, roll_ref=float(entry['roll_angle']))
+
+            # Observational metadata
+            metadata['instrument']['optical_element'] = entry['optical_element']
+            metadata['exposure']['ma_table_number'] = int(entry['ma_table_number'])
+            ma_table = parameters.ma_table[ metadata['exposure']['ma_table_number']]
+
+
+
+
+            # Loop over number of exposures
+            for entry_idx in range(int(entry['exposures'])):
+                # Set initial (possibly only) SCA
+                sca = args.sca if (args.sca > 0) else 1
+
+                # Date
+                date = Time(entry['date'], format='isot') + time_offset
+                metadata['exposure']['start_time'] = date
+
+                # If SCA value is within 1-18, only run once
+                # otherwise loop over all detectors
+                while (sca <= NUMBER_OF_DETECTORS):
+                    # Detector
+                    metadata['instrument']['detector'] = f'WFI{sca:02d}'
+
+                    # Create output file name
+                    output_file_name = f"roman_simulated_{entry['ra']}_{entry['dec']}_{date}_{entry['optical_element']}_{entry_idx + 1}_{sca}.asdf"
+
```

Comment:
Let's leave off the ra/dec/date and just do entry_idx, filter, and sca.  But let's let the user configure that prefix (roman_simulated).  We should also probably include the "suffix" in the name; uncal for l1 and cal for l2.

Reply:
Okay and agreed - the latter is on my TODO list.

Reply:
All three implemented.